sel/plus.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "sel/term.hpp"
+
+#include <cstddef>
+#include <format>
+#include <string_view>
+#include <tuple>
+#include <utility>
+
+namespace sel {
+
+/// associative and commutative n-ary plus operation
+///
+template <term... Args>
+  requires (sizeof...(Args) != 0)
+struct plus
+{
+  std::tuple<Args...> args{};
+
+  constexpr plus(const Args&... args)
+      : args{args...}
+  {}
+
+  friend auto operator<=>(const plus&, const plus&) = default;
```

Comment:
does cpp distinguish some how between partial and total order and equality?

Reply:
To some degree. C++ has different ordering types:
https://en.cppreference.com/w/cpp/utility/compare/strong_ordering
https://en.cppreference.com/w/cpp/utility/compare/weak_ordering
https://en.cppreference.com/w/cpp/utility/compare/partial_ordering.html

Three way compare (i.e., spaceship operator) on floating point types returns a partial_ordering where as ints return a strong_ordering.